packages/provider-elasticsearch/src/utils/highlightUtil.test.js

```diff
@@ -0,0 +1,58 @@
+import { mergeHitHighlights } from './highlightUtil.js'
+
+describe('Highlight field aggregation', () => {
```

Comment:
There are some important test cases we can add here:
1. Test when a highlighted string is contained in another
2. Test when a highlighted string overlaps with another